public/app/plugins/datasource/cloudwatch/components/AnnotationQueryEditor.tsx

```diff
@@ -1,24 +1,23 @@
-import React, { ChangeEvent } from 'react';
-import { PanelData } from '@grafana/data';
+import { QueryEditorProps } from '@grafana/data';
 import { EditorField, EditorHeader, EditorRow, InlineSelect, Space } from '@grafana/experimental';
 import { Input, Switch } from '@grafana/ui';
+import React, { ChangeEvent } from 'react';
 import { CloudWatchDatasource } from '../datasource';
+import { isCloudWatchAnnotationQuery } from '../guards';
 import { useRegions } from '../hooks';
-import { CloudWatchAnnotationQuery, CloudWatchMetricsQuery } from '../types';
+import { CloudWatchJsonData, CloudWatchQuery, MetricStat } from '../types';
 import { MetricStatEditor } from './MetricStatEditor';
 
-export type Props = {
-  query: CloudWatchAnnotationQuery;
-  datasource: CloudWatchDatasource;
-  onChange: (value: CloudWatchAnnotationQuery) => void;
-  data?: PanelData;
-};
+export type Props = QueryEditorProps<CloudWatchDatasource, CloudWatchQuery, CloudWatchJsonData>;
 
-export function AnnotationQueryEditor(props: React.PropsWithChildren<Props>) {
+export const AnnotationQueryEditor = (props: Props) => {
   const { query, onChange, datasource } = props;
-
   const [regions, regionIsLoading] = useRegions(datasource);
 
+  if (!isCloudWatchAnnotationQuery(query)) {
+    return null;
```

Comment:
should this throw an error?

Reply:
hmm yeah this is not ideal. but is throwing errors in the render function better? I think I prefer this, but don't have very strong opinions.

Reply:
this would be an "internal" error, as far as I can tell, users should not hit this "return null".

In that case, it's more likely for them to open an issue (with more useful info) if they see an error in the UI than if the editor is just empty (even though in both cases the editor would be broken).

Reply:
Yeah true but throwing an error in the render function sounds like an anitpattern to me. I'll return an Alert component, ok?

Reply:
see https://github.com/grafana/grafana/pull/48102/commits/8bf636c4434ef96981f430293ceb3b34137767d7

Reply:
yeah, I didn't want to over-complicate this since it probably won't happen but just in case. The Alert should work fine, thanks!